website/server/libs/api-v3/pushNotifications.js

```diff
@@ -11,47 +12,58 @@ let gcm = GCM_API_KEY ? pushNotify.gcm({
   retries: 3,
 }) : undefined;
 
-// TODO review and test this file when push notifications are added back
+let APN_KEY = nconf.get('PUSH_CONFIGS:APN_PEM_FILES:KEY');
 
-if (gcm) {
-  gcm.on('transmitted', (/* result, message, registrationId */) => {
-    // console.info("transmitted", result, message, registrationId);
-  });
+let apn = APN_KEY ? pushNotify.apn({
+  key: nconf.get('PUSH_CONFIGS:APN_PEM_FILES:KEY'),
+  cert: nconf.get('PUSH_CONFIGS:APN_PEM_FILES:CERT')
+}) : undefined;
 
-  gcm.on('transmissionError', (/* error, message, registrationId */) => {
-    // console.info("transmissionError", error, message, registrationId);
+if (apn) {
+  var feedback = apnLib.Feedback({
+    key: nconf.get('PUSH_CONFIGS:APN_PEM_FILES:KEY'),
+    cert: nconf.get('PUSH_CONFIGS:APN_PEM_FILES:CERT'),
+    "batchFeedback": true,
+    "interval": 3600, //Check for feedback once an hour
   });
-
-  gcm.on('updated', (/* result, registrationId */) => {
-    // console.info("updated", result, registrationId);
+  feedback.on("feedback", function(devices) {
+    console.log("GOT FEEDBACK");
+    console.log(devices);
   });
 }
 
-module.exports = function sendNotification (user, title, message, timeToLive = 15) {
-  return; // TODO push notifications are not currently enabled
+module.exports = function sendNotification (user, title, message, identifier, category = "", payload = {}, timeToLive = 15) {
 
   if (!user) return;
+  if (user.preferences && user.preferences.pushNotifications && user.preferences.pushNotifications.unsubscribeFromAll === true) return;
```

Comment:
you can just check for `user.preferences.pushNotifications.unsubscribeFromAll `, the parent objects will always exists